File: backend/src/003-archivos/tipos/csv/manage_csv.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import csv

# ===============================================================

#####   CREAR ARCHIVOS CSV   #####

headers = ["campo1", "campo2", "campo3"]

with open("fichero.csv", "w", newline="") as csvfile:
    writer = csv.DictWriter(csvfile, 
                        fieldnames=headers,
                        restval=' ',
                        quoting=csv.QUOTE_NONNUMERIC)
    # QUOTE_NONNUMERIC convierte todos los campos numéricos en tipo float

    writer.writeheader() # Write header

    insercion = {"campo1": "hola", "campo2": 8, "campo3": "valorfinal"}
    writer.writerow(insercion)

# ===============================================================

#####   LEER ARCHIVOS CSV   #####

with open("fichero.csv", "r", newline="") as csvfile:
    reader = csv.DictReader(csvfile, quoting=csv.QUOTE_NONNUMERIC)

    print(reader.fieldnames)

    for row in reader:  #  ['campo1', 'campo2', 'campo3']
        print(row) # OrderedDict([('campo1', 'hola'), ('campo2', '8'), ('campo3', 'valorfinal')])

# ===============================================================

"""
Fuente:
https://docs.python.org/3/library/csv.html
"""

File: backend/src/003-archivos/tipos/csv/manage_csv.rb
#!/usr/bin/ruby

require 'csv'

# ====================================================

#####   CREAR ARCHIVOS CSV   #####

CSV.open('fichero.csv', 'w') do |csvfile|
    csvfile << ["campo1", "campo2", "campo3"]
    csvfile << ["hola", 8, "valorfinal"]
end

# ====================================================

#####   LEER ARCHIVO CSV   #####

$count, $content = 0, []
CSV.foreach('fichero.csv', col_sep: ',', converters: :numeric) do |row|
    if ($count == 0)
        $headers = row.inspect
    else
        $content.push(row.inspect)
    end
    $count += 1
end

puts $headers, $content

# ====================================================


=begin
Fuentes:
https://www.sitepoint.com/guide-ruby-csv-library-part/
http://ruby-doc.org/stdlib-2.0.0/libdoc/csv/rdoc/CSV.html
=end
